src/Data/Validation/Semigroup.purs

```diff
@@ -56,6 +56,16 @@ isValid _ = false
 toEither :: forall err result. V err result -> Either err result
 toEither (V e) = e
 
+-- | Apply a function if successful, to enable chaining of validation.
+-- |
+-- | Similar to a monadic bind, except it is inconsistent with Apply - that is,
+-- | where as `apply (V err a) (V err a)` accumulates failures,
+-- | `(V err a) ``andThen`` (\a -> V err a)` has fail-fast semantics
```

Comment:
There's a strange mixture of types and terms going on in these examples?
How about:
```purescript
apply (invalid x) (invalid y) = invalid (x <> y)

andThen (invalid x) (\_ -> invalid y) = invalid x
andThen (pure v) b = b v
```